### ISSUE TITLE
arm/neon/aba/native/c and arm/neon/aba/native/cpp fail on ppc64el

### ISSUE THREAD

**Issue by bdrung** [simd-everywhere/simde#1057]
Two test cases fail on ppc64el on Ubuntu 23.04 (mantic) when building from the master branch (as of 2023-08-31):

```
================================== 265/792 ===================================
test:         arm/neon/aba/native/c
start time:   11:32:59
duration:     0.01s
result:       exit status 1
command:      MALLOC_PERTURB_=132 /home/ubuntu/simde/gcc_test/test/arm/neon/aba-native-c
----------------------------------- stdout -----------------------------------
1..12
ok 1 aba/vaba_s8
ok 2 aba/vaba_s16
ok 3 aba/vaba_u8
ok 4 aba/vaba_u16
ok 5 aba/vaba_u32
not ok 6 aba/vaba_s32
ok 7 aba/vabaq_s8
ok 8 aba/vabaq_s16
ok 9 aba/vabaq_u8
ok 10 aba/vabaq_u16
ok 11 aba/vabaq_s32
ok 12 aba/vabaq_u32
----------------------------------- stderr -----------------------------------
../test/arm/neon/aba.c:200: assertion failed: r[0] == simde_vld1_s32(test_vec[i].r)[0] (-1971202570 == -1971202570)

(test program exited with status code 1)
==============================================================================

================================== 266/792 ===================================
test:         arm/neon/aba/native/cpp
start time:   11:32:59
duration:     0.01s
result:       exit status 1
command:      MALLOC_PERTURB_=27 /home/ubuntu/simde/gcc_test/test/arm/neon/aba-native-cpp
----------------------------------- stdout -----------------------------------
1..12
ok 1 aba/vaba_s8
ok 2 aba/vaba_s16
ok 3 aba/vaba_u8
ok 4 aba/vaba_u16
ok 5 aba/vaba_u32
not ok 6 aba/vaba_s32
ok 7 aba/vabaq_s8
ok 8 aba/vabaq_s16
ok 9 aba/vabaq_u8
ok 10 aba/vabaq_u16
ok 11 aba/vabaq_s32
ok 12 aba/vabaq_u32
----------------------------------- stderr -----------------------------------
test/arm/neon/aba.cpp:200: assertion failed: r[0] == simde_vld1_s32(test_vec[i].r)[0] (-1971202570 == -1971202570)

(test program exited with status code 1)
==============================================================================
```

These two test succeed on ppc64el when using the 0.7.6 release (but 126 other arm/neon test cases fail, see [LP #2033648](https://bugs.launchpad.net/ubuntu/+source/simde/+bug/2033648))

**Comment by bdrung** [comment]
Yes, I wanted to see if the tip of the development branch fixed the failure on ppc64el. The test showed that the development branch fixes the failures but regressed on those two test cases.

**Comment by mr-c** [comment]
@bdrung What I'm trying to do is to figure out the source of the failures

1. SIMDE v0.7.6-1 [passes on Debian 'sid'/'testing'](https://ci.debian.net/packages/s/simde/) on pcc64el
   - [gcc 13.2.0-3 libc 2.37-7](https://ci.debian.net/data/autopkgtest/unstable/ppc64el/s/simde/36349569/log.gz)
   - [gcc 13.2.0-1 libc 2.37-6](https://ci.debian.net/data/autopkgtest/unstable/ppc64el/s/simde/36349569/log.gz)
2. SIMDE v0.7.6-1 [fails on Ubuntu mantic](https://autopkgtest.ubuntu.com/packages/simde) on ppc64el
    - [gcc 13.2.0-2ubuntu1 libc 2.38-1ubuntu3](https://autopkgtest.ubuntu.com/results/autopkgtest-mantic/mantic/ppc64el/s/simde/20230826_192116_52513@/log.gz)
3. SIMDE latest dev (v0.7.7+) [passes on Ubuntu Jammy](https://github.com/simd-everywhere/simde/actions/runs/5903001747/job/16012048553?pr=1056) on qemu ppc64el
    - [gcc 12.3.0-1ubuntu1~22.04 libc 2.35-0ubuntu1cross3](https://autopkgtest.ubuntu.com/results/autopkgtest-mantic/mantic/ppc64el/s/simde/20230826_192116_52513@/log.gz)

So I would suggest that this could be a libc 2.38 bug.

**Comment by bdrung** [comment]
Besides comparing the package/library versions, the compiler flags should be compared. Things like `-O2`/`-O3` or LTO could be the culprit. Sadly the logs to not show the compiler flags.

**Comment by mr-c** [comment]
> Besides comparing the package/library versions, the compiler flags should be compared. Things like `-O2`/`-O3` or LTO could be the culprit. Sadly the logs to not show the compiler flags.

We can change the autopkgtests to be verbose, yep!

**Comment by mr-c** [comment]
@bdrung Can you test the recent v0.8.0-rc1 release I made

**Comment by kukrimate** [comment]
@mr-c I've done a test merge with the head of the development branch, and I am getting the following two errors on ppc64el: https://launchpadlibrarian.net/700085671/buildlog_ubuntu-noble-ppc64el.simde_0.8.0~rc1+git+20231129+471a342-0ubuntu1_BUILDING.txt.gz

**Comment by mr-c** [comment]
@kukrimate Thanks for the test! A pretty surprising error ; maybe a compiler error?

```
../test/arm/neon/aba.c:200: assertion failed: r[0] == simde_vld1_s32(test_vec[i].r)[0] (-1971202570 == -1971202570)
```
